### PR TITLE
Revise versioning w.r.t to #409

### PR DESCRIPTION
**Pull request by ezyang**
This PR modifies the operator versioning guidance to permit not revising the version number when there are BC-compatible changes, and defines BC-compatibility to mean that all old operator specifications remain valid with exactly the same semantics.

This specification update affects PRs #397 and #407; I've CC'ed the authors of these PRs since these changes directly affect you. I also CC'ed @linkerzhang and @donbox.

### CODE REVIEW

File: docs/Versioning.md [hunk 1]
```diff
@@ -58,19 +58,46 @@ ISSUE: define type compatibility rules either here or under model versioning - p
...
-Any change of semantics implies a new operator, which MAY share a domain and op_type with another operator. This includes adding new behavior triggered only by previously unspecified inputs or attributes. These changes in semantics also MUST have a distinct operator id.
+The semantics of an operator MAY be extended in a backwards-compatible
+way without requiring the `op_version` of an operator to be increased.
+A backwards compatible semantics change is defined to be a change to
+operator semantics such that all operator definitions which were defined
+in the previous operator definition have exactly same semantics as
+in the previous version.  The following are examples of
+backwards-compatible extensions:
+
+* Adding an optional attribute, such that when the attribute is omitted,
```
Comment (houseroad) [inline docs/Versioning.md]: I like examples. This is much clearer! 👍

File: docs/Versioning.md [hunk 2]
```diff
@@ -58,19 +58,46 @@ ISSUE: define type compatibility rules either here or under model versioning - p
...
+operator id MUST be greater than any extant `op_version` for the
+`domain`.
+
+> In practice, this means that BC-breaking changes in the ONNX
+> repository require contributors to follow the following three steps:
+>
+> 1. Increment the maximum version in `DomainToVersionRange`,
+> 2. Copy the old operator schema to an `old.cc` file, and
+> 3. Update the `SinceVersion` signifier to the new max version from
+>    step (1).
```
Comment (houseroad) [inline docs/Versioning.md]: Shall we also update the version in VERSION_NUMBER file?

Comment (ezyang) [reply docs/Versioning.md]: Nope, VERSION_NUMBER is the PR version number, we only update that when we are releasing onnx proper

Comment (houseroad) [reply docs/Versioning.md]: I see, thanks.

**Pull request merged**